addons/account_edi_ubl_cii/models/account_edi_xml_oioubl_201.py

```diff
@@ -0,0 +1,251 @@
+            })
+        return vals_list
+
+    def _get_invoice_payment_means_vals_list(self, invoice):
+        vals_list = super()._get_invoice_payment_means_vals_list(invoice)
+        for vals in vals_list:
+            # Hardcoded 'unknown' for now
+            # Later on, it would be nice to create a dynamically selected template that would depends on the payment means
+            vals['payment_means_code'] = PAYMENT_MEANS_CODE['unknown']
+
+        return vals_list
+
+    def _get_invoice_tax_totals_vals_list(self, invoice, taxes_vals):
+        vals_list = super()._get_invoice_tax_totals_vals_list(invoice, taxes_vals)
+        for tax_total_vals in vals_list:
+            for subtotal_vals in tax_total_vals.get('tax_subtotal_vals', []):
+                # https://www.oioubl.info/Classes/en/TaxSubtotal.html
+                # No 'percent' node in OIOUBL
+                subtotal_vals.pop('percent', None)
+
+                # TaxCategory https://www.oioubl.info/Classes/en/TaxCategory.html
+                subtotal_vals['tax_category_vals']['id_attrs'] = {
+                    'schemeID': 'urn:oioubl:id:taxcategoryid-1.3',
+                    'schemeAgencyID': DANISH_NATIONAL_IT_AND_TELECOM_AGENCY_ID,
+                }
+
+                # TaxCategory id list: https://www.oioubl.info/codelists/en/urn_oioubl_id_taxcategoryid-1.1.html
+                # This if is necessary because we're coming here several time which would make the result being None
+                if subtotal_vals['tax_category_vals']['id'] not in TAX_POSSIBLE_VALUES:
+                    subtotal_vals['tax_category_vals']['id'] = UBL_TO_OIOUBL_TAX_CATEGORY_ID_MAPPING.get(subtotal_vals['tax_category_vals']['id'])
+
+                subtotal_vals['tax_category_vals']['tax_scheme_name'] = 'VAT'
+                subtotal_vals['tax_category_vals']['tax_scheme_id_attrs'] = {'schemeID': 'urn:oioubl:id:taxschemeid-1.5'}
+
+                # /Invoice[1]/cac:TaxTotal[1]/cac:TaxSubtotal[1]/cac:TaxCategory[1]
+                # [W-LIB230] Name should only be used within NES profiles
+                # (cbc:Name != '') and not(contains(/doc:Invoice/cbc:ProfileID, 'nesubl.eu'))
+                if 'name' in subtotal_vals['tax_category_vals']:
+                    del subtotal_vals['tax_category_vals']['name']
+
+        return vals_list
+
+    def _get_legal_monetary_total_vals(self, invoice, taxes_vals, line_extension_amount, allowance_total_amount):
+        vals = super()._get_legal_monetary_total_vals(invoice, taxes_vals, line_extension_amount, allowance_total_amount)
+        # In OIOUBL context, tax_exclusive_amount means "tax only"
+        vals['tax_exclusive_amount'] = taxes_vals['tax_amount_currency']
+        if invoice.currency_id.compare_amounts(vals['prepaid_amount'], 0) <= 0:
+            del vals['prepaid_amount']
+        return vals
+
+    def _get_invoice_payment_terms_vals_list(self, invoice):
+        # TODO BIB: implement cac:PaymentTerms
```

Comment:
Did you create a task for it? (I know todos are worth nothing :D )

Reply:
Indeed :D , I will create a task so it doesn't get forget.